src/REPLICA/fix_pimd_langevin.cpp

```diff
@@ -0,0 +1,1497 @@
+  delete[] mass;
+  delete[] _omega_k;
+  delete[] Lan_c;
+  delete[] Lan_s;
+  delete[] tau_k;
+  delete[] c1_k;
+  delete[] c2_k;
+  delete[] plansend;
+  delete[] planrecv;
+  delete[] modeindex;
+  memory->destroy(xcall);
+  if (cmode == SINGLE_PROC) {
+    memory->destroy(bufsorted);
+    memory->destroy(outsorted);
+    memory->destroy(bufsortedall);
+    memory->destroy(buftransall);
+    memory->destroy(counts);
+    memory->destroy(displacements);
+  }
+
+  if (cmode == MULTI_PROC) {
+    memory->destroy(bufsendall);
+    memory->destroy(bufrecvall);
+    memory->destroy(tagsendall);
+    memory->destroy(tagrecvall);
+    memory->destroy(counts);
+    memory->destroy(displacements);
+  }
+  memory->destroy(M_x2xp);
+  memory->destroy(M_xp2x);
+  memory->destroy(xc);
+  memory->destroy(x_unwrap);
+  memory->destroy(bufsend);
+  memory->destroy(bufrecv);
+  memory->destroy(tagsend);
+  memory->destroy(tagrecv);
+  memory->destroy(bufbeads);
+}
+
+/* ---------------------------------------------------------------------- */
+
+int FixPIMDLangevin::setmask()
+{
+  int mask = 0;
+  mask |= POST_FORCE;
+  mask |= INITIAL_INTEGRATE;
+  mask |= FINAL_INTEGRATE;
+  mask |= END_OF_STEP;
+  return mask;
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixPIMDLangevin::init()
+{
+  if (atom->map_style == Atom::MAP_NONE)
+    error->all(FLERR, "fix pimd/langevin requires an atom map, see atom_modify");
+
+  if (universe->me == 0 && universe->uscreen)
+    fprintf(universe->uscreen, "fix pimd/langevin initializing Path-Integral ...\n");
+
+  // prepare the constants
+
+  masstotal = group->mass(igroup);
+  inverse_np = 1.0 / np;
+
+  double planck;
+  if (strcmp(update->unit_style, "lj") == 0) {
+    double planck_star = sqrt(lj_epsilon) * sqrt(lj_mass) * lj_sigma * sqrt(other_mvv2e);
+    planck = other_planck / planck_star;
+  } else {
+    planck = force->hplanck;
+  }
+  planck *= sp;
+  hbar = planck / (2.0 * MY_PI);
+  double beta = 1.0 / (force->boltz * temp);
+  double _fbond = 1.0 * np * np / (beta * beta * hbar * hbar);
+
+  omega_np = np / (hbar * beta) * sqrt(force->mvv2e);
+  beta_np = 1.0 / force->boltz / temp * inverse_np;
+  fbond = _fbond * force->mvv2e;
+
+  if ((universe->me == 0) && (universe->uscreen))
+    fprintf(universe->uscreen,
+            "fix pimd/langevin -P/(beta^2 * hbar^2) = %20.7lE (kcal/mol/A^2)\n\n", fbond);
+
+  if (integrator == OBABO) {
+    dtf = 0.5 * update->dt * force->ftm2v;
+    dtv = 0.5 * update->dt;
+    dtv2 = dtv * dtv;
+    dtv3 = THIRD * dtv2 * dtv * force->ftm2v;
+  } else if (integrator == BAOAB) {
+    dtf = 0.5 * update->dt * force->ftm2v;
+    dtv = 0.5 * update->dt;
+    dtv2 = dtv * dtv;
+    dtv3 = THIRD * dtv2 * dtv * force->ftm2v;
+  } else {
+    error->universe_all(FLERR, "Unknown integrator parameter for fix pimd/langevin");
+  }
+
+  comm_init();
+
+  mass = new double[atom->ntypes + 1];
+
+  nmpimd_init();
+
+  Langevin_init();
```

Comment:
For consistency, please rename this function to all lowercase (`langevin_init()`). See my other comment.

Reply:
solved by 
a415d73